src/utils/daos_metrics/daos_metrics.c

```diff
@@ -165,9 +158,18 @@ main(int argc, char **argv)
 		}
 	}
 
+	if (format == D_TM_CSV)
+		filter &= ~D_TM_DIRECTORY;
+	else
+		filter |= D_TM_DIRECTORY;
+
 	while ((num_iter == 0) || (iteration < num_iter)) {
-		print_my_children(shmem_root, root, filter | D_TM_DIRECTORY, 0,
-				  stdout);
+		d_tm_print_my_children(shmem_root, root, 0,
+				       filter, dirname,
+				       show_meta, format, stdout);
+		if (format == D_TM_CSV)
+			break;
```

Comment:
The use model I had in mind was that a person might actually look at the standard output tree, and want to see multiple iterations as they are looking in real time as the data changes.  For the CSV use model, I thought that someone would only want to dump the CSV data to a file to capture the final state after a run.  In that case, iterating didn't make a lot of sense.  That said, there is no reason why it has to be limited in CSV mode.  So I can remove that restriction if we want.   If we do that, I'm wondering if we need to put a sequence number in the CSV data, or some other way to delineate one iteration from another?

Reply:
I think it would make sense to number the iterations in the CSV output. If the default num_iter == 1, then the user would have to specify different behavior if they really want it.

Reply:
I made some modifications here, and decided to go with a timestamp field instead of iteration number.  I thought that was ultimately more helpful.  The iteration number can be gleaned from that if someone wants to.